odl/tomo/geometry/geometry.py

```diff
@@ -0,0 +1,209 @@
+        normalized : `bool`, optional
+            If `True`, return a normalized (unit) vector. Default: `True`
+
+        Returns
+        -------
+        vec : `numpy.ndarray`, shape (`ndim`,)
+            (Unit) vector pointing from the detector to the source
+        """
+
+    def src_position(self, mpar):
+        """The source position function.
+
+        Parameters
+        ----------
+        mpar : element of motion parameters `motion_params`
+            Motion parameter for which to calculate the source position
+
+        Returns
+        -------
+        pos : `numpy.ndarray`, shape (`ndim`,)
+            The source position, a `ndim`-dimensional vector
+        """
+        raise NotImplementedError
+
+    def det_point_position(self, mpar, dpar):
+        """The detector point position function.
+
+        Parameters
+        ----------
+        mpar : element of motion parameters `motion_params`
+            Motion parameter at which to evaluate
+        dpar : element of detector parameters `det_params`
+            Detector parameter at which to evaluate
+
+        Returns
+        -------
+        pos : `numpy.ndarray`, shape (`ndim`,)
+            The source position, a `ndim`-dimensional vector
+        """
+
+        # TODO: check and write test
+        return np.asarray(
+            (self.det_refpoint(mpar + self.motion_params_offset) +
+             self.det_rotation(mpar).dot(self.detector.surface(dpar))))
+
+    @property
+    def motion_params_offset(self):
+        """Offset to the motion parameters `motion_params`."""
+        return None
+
+    @property
+    def motion_grid(self):
+        """A sampling grid for `motion_params`."""
+        return None
+
+    @property
+    def has_motion_sampling(self):
+        """Whether there is a `motion_grid` or not."""
+        return self.motion_grid is not None
+
+    @property
+    def det_params(self):
+        """The detector parameters."""
+        return self.detector.params
+
+    @property
+    def det_grid(self):
+        """A sampling grid for `det_params`."""
+        return self.detector.param_grid
+
+    @property
+    def has_det_sampling(self):
+        """Whether there is a `det_grid` or not."""
+        return self.det_grid is not None
+
+    @property
+    def params(self):
+        """Joined motion and detector parameters."""
+        params = IntervalProd([], [])
+        params = params.insert(self.motion_params, params.ndim)
+        params = params.insert(self.det_params, params.ndim)
+        return params
+
+    @property
+    def grid(self):
+        """Joined sampling grid for motion and detector parameters."""
+        if (isinstance(self.motion_grid, RegularGrid) and
+                isinstance(self.det_grid, RegularGrid)):
+            grid = self.motion_grid
+        else:
+            grid = TensorGrid(*self.motion_grid.coord_vectors)
+        return grid.insert(self.det_grid, grid.ndim)
```

Comment:
Well to me, this should hold (rg = regular, tg = tensor)

```
rg.insert(rg) --> rg
rg.insert(tg) --> tg
tg.insert(tg) --> tg
tg.insert(tg) --> tg
```

Thus this would solve itself (if we fixed this).